### ISSUE TITLE
Support ssh client configs

### ISSUE THREAD

**Issue by netdisciple** [bloomberg/powerfulseal#62]
With the current implementation of spur lib for ssh connectivity, I do not see a way for providing an ssh client config (whereby one may have a client config that uses a ProxyCommand configuration). Please consider supporting ssh client config (rather than passing a path to a private key), as modern access security models support multiple keys for access to inside hosts through bastion/edge nodes.

**Comment by netdisciple** [comment]
yep, fabric is the way to go. however, i did find the reference to "sock=paramiko.proxy.ProxyCommand" buy its yucky this way and doesn't solve the problem elegantly. these params should be buried in the client config, not in code.

**Comment by kz** [comment]
Implementing Fabric might lead to a minor feature removal as Fabric does not support adjusting whether unknown hosts are allowed. Will have to see whether this is an acceptable change. (https://github.com/fabric/fabric/issues/1804)
